### ISSUE TITLE
Possible bug with using Linux Azure Functions and get_stations not working

### ISSUE THREAD

**Issue by jpdc-developer** [obspy/obspy#2745]
# Issue and Example: 

I am creating an FDSN client to use the get_stations function to get the stations.

To do this, I am calling get_stations as follows:

```
  inventory = client.get_stations(latitude = latitude , longitude = longitude,
                                                maxradius = maxradius , starttime = starttime,
                                                endtime = endtime, location ="2*", channel="H*", level="response")
```

This runs fine locally and on windows. After deploying it to a Python Azure Function running on Linux, I am getting the following error:

```
 in get_relevant_stations inventory = client.get_stations(latitude = site_latitude,
 longitude = site_longitude,
 File "/home/site/wwwroot/.python_packages/lib/site-packages/obspy/clients/fdsn/client.py",
 line 732,
 in get_stations inventory = read_inventory(data_stream) File "<decorator-gen-49>",
 line 2,
 in read_inventory File "/home/site/wwwroot/.python_packages/lib/site-packages/obspy/core/util/decorator.py",
 line 300,
 in _map_example_filename return func(*args,
 **kwargs) File "/home/site/wwwroot/.python_packages/lib/site-packages/obspy/core/inventory/inventory.py",
 line 88,
 in read_inventory return _generic_reader(path_or_file_object,
 _read,
 format=format,
 File "/home/site/wwwroot/.python_packages/lib/site-packages/obspy/core/util/base.py",
 line 677,
 in _generic_reader generic = callback_func(fh.name,
 **kwargs) File "<decorator-gen-50>",
 line 2,
 in _read File "/home/site/wwwroot/.python_packages/lib/site-packages/obspy/core/util/decorator.py",
 line 210,
 in uncompress_file result = func(filename,
 *args,
 **kwargs) File "/home/site/wwwroot/.python_packages/lib/site-packages/obspy/core/inventory/inventory.py",
 line 97,
 in _read inventory,
 format = _read_from_plugin('inventory',
 filename,
 format=format,
 File "/home/site/wwwroot/.python_packages/lib/site-packages/obspy/core/util/base.py",
 line 450,
 in _read_from_plugin raise TypeError('Unknown format for file %s' % filename)Result: Failure Exception: TypeError: Unknown format for file /tmp/obspy-0yop00ed.tmp 
```

I think there's an issue with ObsPy and Linux, or perhaps more specifically  ObsPy and Azure Functions running on Linux. Is this a bug to be fixed, and are there any known workarounds?

I’ve tried saving the API response in a file by specifying a filename and type. I can save this file successfully. I can also read this file manually using Python, but using ObsPy’s read_inventory function still raises the error. Both saving the results into a file, and into an object, work in my local Windows environment though.

I've checked the open and closed issues with get_stations and I can't find an answer to the issue that I'm having

# Environment:
  -  ObsPy version = 1.2.2, Python version = 3.8.6  and Platform Linux (Azure Function)
  -  Installed ObsPy using Pip, Python using the downloadable installer

**Comment by calum-chamberlain** [comment]
This looks similar to the google colab issues (#2547 and #2697) which relate to needing to reload the environment to get the plugins used for reading to register.

I haven't used Azure much, but is there a way to "restart the runtine" as the fix is for the google colab issues?

**Comment by jpdc-developer** [comment]
> This looks similar to the google colab issues (#2547 and #2697) which relate to needing to reload the environment to get the plugins used for reading to register.
> 
> I haven't used Azure much, but is there a way to "restart the runtine" as the fix is for the google colab issues?

Thanks for the response. Azure Functions gives you limited control over the environment but I'll try various things to "restart the runtime". 

Can you please expand on what it means to restart the runtime? Does it mean restarting the python process like in:

https://stackoverflow.com/questions/53154369/google-colab-how-to-restart-runtime-using-python-code-or-command-line-interf

**Comment by calum-chamberlain** [comment]
With google colab there is a "Runtime" button at the top of the notebook that has an option to restart - this was the suggested fix for the colab issue. 

I am unsure whether running `os.kill(...)` (as suggested in the stackoverflow link you sent) would work in the same way.

**Comment by ThomasLecocq** [comment]
See also the recent discussions on gitter

**Comment by jpdc-developer** [comment]
> See also the recent discussions on gitter

Which discussions are you referring to? I've visited https://gitter.im/obspy/obspy

**Comment by jpdc-developer** [comment]
> With google colab there is a "Runtime" button at the top of the notebook that has an option to restart - this was the suggested fix for the colab issue.
> 
> I am unsure whether running `os.kill(...)` (as suggested in the stackoverflow link you sent) would work in the same way.

I've tried 2 to restart the runtime (which didn't work):

- I made another function in the same Python Azure Function on Linux which executes os.kill. I ran that function manually using the test interface on Azure, then I ran the "main: function running ObsPy. That didn't fix the issue.

- There' no real option to restart the function, on the current portal, but there is a command that can restart the function using the Azure CLI:

```
az functionapp restart --name <functionappName> --resource-group <resourceGroup>
```

I ran it in verbose mode to confirm it did something and then ran the main function again. I'm still experiencing the error.

Is there anything else I can try to work around this? Perhaps this needs a code fix in the library?
